src/Workspaces/Core/Portable/Storage/SQLite/v2/Interop/SqlConnection.cs

```diff
@@ -231,34 +239,39 @@ public void RunInTransaction<TState>(Action<TState> action, TState state)
                 ExecuteCommand("begin transaction");
                 var result = action(state);
                 ExecuteCommand("commit transaction");
-                return result;
+                return (result, null);
             }
-            catch (SqlException ex) when (ex.Result is Result.FULL or
-                                          Result.IOERR or
-                                          Result.BUSY or
-                                          Result.LOCKED or
-                                          Result.NOMEM)
+            catch (SqlException ex)
```

Comment:
we were previously special casing these so we would clean up carefully.  however, we would then still continue throwing these upwards where they would fail out and cause NFWs.  

The new approach is to:

1. always clean up. (always a good idea, and necessary for those cases anyways)
2. bubble up failures for readers as *thrown* exceptions.  we can see if we continue to get NFWs when we do that.
3. bubble up failures for writers as *returned* exceptions.  the writers can determine what should be done in those cases.